### ISSUE TITLE
Explore Routes that span multiple neighborhoods

### ISSUE THREAD

**Issue by misaugstad** [ProjectSidewalk/SidewalkWebpage#3488]
##### Brief description of problem/feature
To speed up implementation, we made it so that RouteBuilder only lets you explore routes that are contained to a single neighborhood. It's a weird restriction from a user's perspective, so we should work on how to remove the restriction!

It's complicated because all the way up the chain, we assume that you are exploring in a single neighborhood. We show a map of the neighborhood on mission complete, there are checks throughout the Explore page code to see if you've completed the neighborhood you're in, the `mission` table has a `region_id` field, we scale the length of your missions based on how many you've completed in the given neighborhood, etc.

It's certainly doable to make this change, it was just worthwhile to handle it completely separately from the initial RouteBuilder release.

**Comment by jonfroehlich** [comment]
Resurfaced hard during RouteBuilder QA today, and worth capturing because the single-neighborhood confinement now fails in a way that's invisible to users.

**Today's symptom:** on a route whose start snapped to a **boundary node where two neighborhoods' street-edges share the exact same endpoint (0 m apart)** — teaneck node `40.88819, −74.00515`, where region-15 edge `1201` and region-21 edge `1202` coincide — you couldn't place an End *anywhere*. `RouteGraph.route()` re-snapped the already-placed Start without a region filter, the 0 m tie resolved to the *other* neighborhood, and every End then failed the `from.regionId !== to.regionId` check — shown as the generic "not connected by a walkable path." No "can't-place" cursor fires, because the point *is* inside a valid region polygon; the mismatch only surfaces inside routing.

**Stopgap just landed** (RouteGraph.route now snaps both endpoints within the route's locked region, so a boundary node resolves deterministically): unblocks single-neighborhood routes at boundaries. It does **not** lift the confinement — that's this issue.

**The real fix is more tractable than when this was filed.** The RouteBuilder route-scoped-missions work (#4579) already decoupled several of the blockers listed above:

| #3488 blocker | Now |
|---|---|
| `mission` assumes single neighborhood | route walks use one **route-scoped mission** (`mission.user_route_id`), sized to the route |
| mission length scaled by neighborhoods completed | bypassed for routes — mission = route length |
| "have you completed the neighborhood" checks | route completion is route-driven (ends at the route's last reachable pano) |
| mission-complete shows the neighborhood map | modal already branches for routes |

**Remaining to lift confinement to deployment (city) scope** — the sensible boundary for a first-class route, vs. neighborhood:

1. `route` / `user_route` currently pin a single `region_id` — allow a route to span regions (nullable / "primary" region, or drop the coupling).
2. RouteBuilder routing: remove the per-region A* filter + one-neighborhood build guard (`RouteGraph.route` region checks, the `one-neighborhood-warning`), keeping routes within the **deployment** boundary.
3. Explore walking: current-region assignment as a walker crosses a neighborhood; `mission.region_id` for a multi-region route.
4. Keep it its own PR — it reaches route storage + current-region + task loading and deserves dedicated QA.

**Comment by jonfroehlich** [comment]
## Investigation: removing the "region requirement" (2026-09-18)

Goal, per discussion with @jonfroehlich: keep regions as the unit that organizes missions, but stop (1) cutting streets at region borders and (2) forcing routes to be self-contained in one region. Streets and routes should make sense on the ground and end at intersections, not at an invisible polygon line. Related: #4717, #3987, #624.

### The requirement is two separate couplings

**1. Streets are physically cut at region borders.** This happens in exactly one place: the `gpd.overlay(..., how='intersection')` in `assign_regions` in `scripts/onboard_city.py`. Five healing functions (`bridge_short_gaps`, `restore_boundary_tails`, `heal_edge_pieces`, `coalesce_pieces`, `absorb_boundary_riders`) then try to undo the damage. Nothing re-cuts later; the OSM refresh only touches tags.

**2. Routes are confined to one region.** Enforced only in the browser: three region filters in `RouteGraph.js` (`snapToStreet`, the `different-region` error, the A* edge filter) plus the region lock in `RouteBuilder.js`. It is backed by `route.region_id NOT NULL`, which `ExploreService` reads to pin the whole walk to one region. The server never validates it, and `updateRoute` already lets `region_id` go stale when a route's streets are replaced.

Everything else only needs each street to be **assigned** to one region. Missions, task selection, `region_completion`, the public API and AccessScore all join through `street_edge_region`, which is UNIQUE on `street_edge_id` (evolution 338). None of it needs the street's geometry to lie inside the polygon. The Explore client never reads the region polygon: `Neighborhood.getGeoJSON()` has no callers and there is no containment check anywhere. So the 1:1 assignment stays, the geometric guarantee goes, and **no schema change is needed**.

### What the cutting buys (Seattle, open streets, geodesic lengths)

A "cut" = two open street edges from the same OSM way, in different regions, touching end to end.

| | |
|---|---|
| Border cuts | 1,603 |
| Cuts whose smaller piece is < 20 m | 1,303 (81%) |
| Median smaller piece | 1.8 m |
| p90 smaller piece | 65.9 m |
| Cuts leaving ≥ 200 m on both sides | 22 (1.4%) |
| Sub-20 m streets that are direct products of a cut | 1,408 of 3,352 |
| Length that changes region if every cut were undone | 29.8 km (1.3% of the network) |
| …if only the < 20 m slivers were undone | 3.5 km |

The concern in #3987 — an uncut street running half a mile into the next neighborhood — applies to about 1% of cuts. Streets are already noded at intersections, so an uncut street overhangs its region by at most one block. City-wide totals stay exact either way; only the per-region split moves.

### Plan: three independent PRs

**PR 1 — routes span regions (this issue).** Pure code, no data migration, so it goes first.

- *RouteBuilder:* remove the region lock, the `different-region` error, the A* region filter, the `one-neighborhood-warning` string and the select-a-neighborhood stage. Clicking a neighborhood becomes a zoom aid. The boundary-node tie bug described above disappears, because A* can traverse both coincident edges. Auto-routing now runs *through* slivers, which removes most of the #3987 pain before any merge happens.
- *`route.region_id`:* keep the column, redefine it as the route's **start region**, and derive it server-side from the first street in both `saveRoute` and `updateRoute` (which also fixes the stale-region bug). No evolution. Listings show the start region plus a neighborhood count.
- *Explore, server:* a route walk uses the region of the current task's street for the page payload and leaves `user_current_region` alone, as the address drop-in already does. A route mission's `mission.region_id` stays the start region.
- *Explore, client:* three region-scoped calls need route-scoped variants — the resume-labels fetch (today a user's earlier labels on out-of-region streets would not reload and could be re-placed as duplicates), the live street-priority refresh, and `/neighborhoodMissions`.
- *Unchanged:* walking, compass, minimap and progress are already route-ordered and region-blind; `region_completion` already credits each street to its own region.

**PR 2 — merge existing cuts (#4717 Phase 1, tier 1).** A data repair across ~50 cities, staged localhost → test → prod. Scope to settle there: slivers only (< 20 m minority piece, 81% of cuts) versus every same-way cut.

**PR 3 — stop cutting in new cities.** Replace the overlay and the five healing functions with a clip to the city coverage plus largest-overlap region assignment (ties to the lowest region id — the rule the DC migration already used). Long overhangs go to a QA layer. Reword the API doc string saying a label's region is the one it "falls within"; it is the region of the label's street.

### Decisions made

1. New cities: **never cut** at region borders.
2. A route mission's `mission.region_id`: **the start region**.
3. Start with PR 1.

### Side effects to expect

- Per-region distance totals, AccessScore region weights and label-by-region counts credit a whole street to its one region, so a border street's metres and labels land entirely in the majority region. This is how the code already behaves; uncut streets just make it visible at borders (≈1.3% of Seattle's network at most).
- On region maps, some labels will sit slightly outside their region's polygon, along streets that ride a border.

### Methodology

Read-only SELECTs against a local copy of the Seattle schema; `status = 'open'` streets; lengths via `ST_Length(geom::geography)`; touching = within ~1 m. Code references verified against `develop` at 98a1d517e.

🤖 Generated with [Claude Code](https://claude.com/claude-code) — Fable 5.1, claude-fable-5-1

**Comment by misaugstad** [comment]
>PR 3 — stop cutting in new cities. Replace the overlay and the five healing functions with a clip to the city coverage plus largest-overlap region assignment (ties to the lowest region id — the rule the DC migration already used). Long overhangs go to a QA layer. Reword the API doc string saying a label's region is the one it "falls within"; it is the region of the label's street.

I'm not convinced that we need this step... I think that the adjustments that I made (the "Five healing functions") probably get us all that we need here. As was said earlier, "an uncut street running half a mile into the next neighborhood — applies to about 1% of cuts", so with these healing functions, I think that cuts are not super common anymore.

And the proposed alternative, "Long overhangs go to a QA layer", just takes the part that I've automated and puts it back into a manual process.

I'm happy with the outline for PRs 1 and 2 though! I'd aim PR 2 to try to match what we do in the automatic city onboarding closely, as I did quite a bit of QA on it, and then cities will match each other more closely!
